compiler/luci/pass/src/QuantizationUtils.cpp

```diff
@@ -395,4 +399,38 @@ void quant_const(luci::CircleConst *node, loco::DataType quant_type)
   node->quantparam(std::move(quantparam));
 }
 
+namespace
+{
+
+// TODO move this to a more global helper file
+int nbits(loco::DataType dt) noexcept
+{
+  switch (dt)
+  {
+    case loco::DataType::S8:
+    case loco::DataType::U8:
+      return 8;
+    case loco::DataType::S16:
+    case loco::DataType::U16:
+    case loco::DataType::FLOAT16:
+      return 16;
+    case loco::DataType::S32:
+    case loco::DataType::U32:
+    case loco::DataType::FLOAT32:
+      return 32;
+    case loco::DataType::S64:
+      return 64;
+    default:
+      return 64; // a safe large default
+  }
+}
+
+} // namespace
+
+bool range_check(float min, float max, loco::DataType dtype)
+{
+  float thresh = 1.2f;
+  return log2f(max) - log2f(min) > nbits(dtype) * thresh;
```

Comment:
Do you have any reference for this metric?

`min/max` is changed or ignored according to quantization schemes. So, how about using `scale` rather than `min/max`? Scale is the result of quantization algorithms, thus independent of quantization schemes. In addition, I think it would be possible to make warnings when quantizing activation.

If you'll make warnings only for const nodes, I think that quantization error (e.g., mean percentage error of `|original weights - quantized/dequantized weights|`) is a better proxy to evaluate accuracy.

Reply:
It's what I used to find a bug with a customer model. 
`mean percentage error of |original weights - quantized/dequantized weights|` is a good metric, but it's more expensive to compute (O(N) vs O(1)).
There's a tradeoff here: how often do we want to warn the user? The more numerous warnings are more easily ignored, because they are perceived to have lower signal to noise ratio.

Reply:
I see. Then could you add TODO comment like,

'TODO Check if the metric is proper'